### PR TITLE
[Merged by Bors] - update demos for 23.7

### PR DESCRIPTION
**Pull request by adwk67**
## Description

Fixes https://github.com/stackabletech/issues/issues/402

N.B. DO NOT MERGE UNTIL AFTER RELEASE-23.7!!
Keep this PR open until after the release so that e.g. versions 0.0.0-dev can be replaced with 23.7

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: stacks/trino-iceberg/trino.yaml [hunk 1]
```diff
@@ -46,16 +43,23 @@ spec:
...
 type: kubernetes.io/opaque
 stringData:
-  # htpasswd -n -B -C 10 admin
-  # Hack: Trino requires bcrypt version 2Y, the bcrypt crate generates 2B hashes. See https://github.com/trinodb/trino/issues/11314 for details
-  # This can be removed once the bcrypt crate starts generating 2Y hashes
-  admin: "{{ bcrypt(password=trinoAdminPassword) | replace(from="$2b$", to="$2y$") }}"
+  admin: "{{ trinoAdminPassword }}"
 ---
 # As the demo needs to have access to Trino using the admin user we need to store the password somewhere
 apiVersion: v1
```
Comment (sbernauer) [inline stacks/trino-iceberg/trino.yaml:65]: Don't forget to remove all `trino-credentials` Secrets as well as all references to it ;)

Comment (adwk67) [reply stacks/trino-iceberg/trino.yaml]: Thx! (done)

**Review by sbernauer — APPROVED** [review]
LGTM

**Pull request closed without merging**